### DISCUSSION TITLE
Meta_Hours_7

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/nft-gallery#2737] [Meta-Hours]
![image](https://user-images.githubusercontent.com/5887929/161279314-b4569827-cb9d-41c6-8f1f-e24491263beb.png)

> The Grid
A digital frontier
I tried to picture clusters of information as they moved through the computer
What did they look like? Ships? Motorcycles?
Were the circuits like freeways?
I kept dreaming of a world I thought I'd never see
And then one day
I got in

We will have a fair share to share. 
From Forming Builders developer guild to accelerating KodaDot to other universes.
Keep it coming.

# Discord event 
- https://discord.com/events/840514076538830888/959451836971106364

Get into mood 
- https://open.spotify.com/playlist/49IoHZIsKiAOPqfGhaWkTv?si=Jo_YQwYQTmWqniwCg7B0HA
- https://open.spotify.com/playlist/1slK5Yq2LE5OiywDd9Mqyl?si=-3OtiL4aR5q02FQmgja4AA
- https://open.spotify.com/playlist/4vE7EXa9RsSw08qLwL4n8M?si=joNuoYQ0RBWBzy4vx5AJaw
- https://open.spotify.com/playlist/0lHPoFzQs3g9ncnsEQLrSr?si=rpt_pu5oQZmd3J2safYthg
- https://open.spotify.com/playlist/0nw7OOXTi1KEgOp2lqTQcG?si=T3lXZ9P6THeeJ1TrUdn0jQ
- https://open.spotify.com/playlist/5GEj4G4NobMLGnxgDhOEz6?si=dSxFHeL5T4aHCx8EMivZbw

### Weights which issue to pick
- I guess this is definitely thought one, but let me try to give my best guess and bet. We have labels like p1-p5. p1 is super broken for users, p2 we would like to see in production under 2 weeks well tested, p3 is there and would be good to be done in 1 month approx. p4 are we considered to have it and decided to work on that. p5 are mostly there in status quo requires further research but we have intent to have something around that issue.
- Another significant historical move would be to look at our issues numbered under 1000. There are still some leftovers from year-long or August-October as there might be definitely strategic issues to tackle. 
- I'm aware that some issues are duplicates or very similar, but let's zoom out, find a similar one and **let's link them together** to be our SEO expert on issues for developers, you are helping each other orient and coordinate make seamless. If some issues are repeated, means that we really care about that "feature/upgrade" and for some reason, it was amplified in our discussion and we noted it down :)
- We've got recently introduced milestones where I've highlighted issues that would be good to tackle, it's still flexible and I might try to add others as I've noticed which would be strategically important to have in KodaDot in this state 
- #2622 
- #2646 

### Forming Developers Guild
As we spoke recently on all hands dev call #2 we would like to form a developers guild, it fits our thesis for the future of KodaDot so I will be exploring ways and what other developer DAOs out there how to incentivize and to reward builders, and what is perfect optimal process to drive development in the most possible and fastest way :) 

### Review bot

Hence we are getting KodaDot Code Review Guild, we will try round-robin or load balance between reviewers' upcoming PRs to speed up merging and break the bottleneck by distributing and engaging long-term contributors. 
I'll be looking forward to figuring out a fair model similar to paying out based on merged PRs 
- https://github.com/kodadot/nft-gallery/issues/2730

### I'm happy
I'm quite happy since stepped out from major reviewing PRs and handing over payouts to long-term contributors. This leaves me more room to layout missing strategical decisions and back to the hiring pipeline to grow our contributor's team and faster-forming builders Guilds around KodaDot, I'm quite excited to see what we will look like in 3-6 months from now as it's becoming mayor milestone for the whole project.

### Development is progressing 
Hence I've installed repography we can clearly see that our progress seems taking a steep turn with reviewers' by code review guild support :) @kodadot/code-review-guild 

![image](https://user-images.githubusercontent.com/5887929/163174602-9559463f-85cb-4c16-8330-ec2e969d5c28.png)
![image](https://user-images.githubusercontent.com/5887929/163174619-c4689b5f-74aa-43bc-8186-d98e40456d54.png)

### AmsterDot
We will be presenting [Amsterdot.io](https://amsterdot.io) where @vikiival give a talk about Reactive NFTs among the top and legitimate Dotsama projects, I'm pretty stoked to show up there and see you all IRL, as some of you we will be meeting after 9months+ or so :)

### Sustainability
As been addressed, people might have concerns about our long-term funding and sustainable revenue stream to cover upcoming **maintenance costs**. With OSS it's definitely though one. Important part we exited pit of one-man-show and we have lot of contributors, on monthly average 16 folks which is I would say amazing. 

We are actively looking for ways to **fund KodaDot as a public good**, for example through Kusama Treasury initially to test it out and if we are proven good to go and well spend resources, I bet it could continue on a recurring basis with some **feedback tweaks and high-quality code base reusability.** Other than that one old school model is Whitelabel, I'm taking this also into consideration where we could accrue fees from particular projects where they are in pipeline. 

Another model why we started building KodaDot in the way we do is mutual beneficial integrations with other parachains, where we help them leverage their **original NFT functionality and provide liquidity to users.**. Including seamless teleportation which will be ofc though question I bet with partners like Composable it would be an easy lift. 

First, the upcoming parachain integration is Basilisk, where we have significantly contributed our know-how around NFTs and they built the NFT marketplace with the help of their Rust developers. Despite locked balances, we anticipate the launch of their NFT marketplace quite soon. What's the inside box? We would be happy to introduce **royalties,  make offers and proxy royalties** as signature features where so far no one has royalties and offers it in the ecosystem in a trustless way. Singular (RMRKv2) **has a centralized version just for the record and has nothing to do with smart contracts nor pallets in the current state.** 

To be fair and finish this take on short-term sustainability, by [looking back to the leaderboard](https://github.com/kodadot/nft-gallery/blob/main/LEADERBOARD.md) and [burn_rate](https://github.com/kodadot/nft-gallery/issues/2642), we already spend some fair amount money which has on efficiency paid quite well comparing other project burn rates. To assure you, we've reserved a fair amount for development approx 100k from current funding, where the original funding we got should go on the development of our very own experimental NFT parachain Metaprime, we will fairly extend this founding 3-8x depending on how well our upcoming fundraising will go. 

To wrap up things, we have a fair funding runway for upcoming months for sure with a long-term vision of 3-5 years until KodaDot will start covering its own maintenance and integration costs, which I bet will happen even sooner if we will try speed-up things.

# MetaPrime time

https://user-images.githubusercontent.com/5887929/163140373-0c1a2d76-d47f-457d-98e3-95357100bb1a.mov

https://nemutas.github.io/r3f-flowmap/
ht @vikiival pic https://twitter.com/focru_ino/status/1513194740321521664?s=21&t=i159Z84fRLf3aGhnPFZ1Yw

Once I've got back time in my back office and regenerated my brain muscles to drive the extra cognitive heavy lift. We are starting brewing our KodaProtocol and MetaPrime testnet. We've been all through weird quirks people invented in past since NFTs are just digital collectables from the 90s, yet now easier to "collect" and send value to the creator. That's the creator's economy.

Okay, you might ask what we've done so far on this page? We've **done tons of research on what we should implement **natively** into MetaPrime** to carry our experimental NFT landscape further, having cheaper interactions and accommodating most needs from artists' culture and recent steady rising GameFi culture.

- We are fairly getting to some answers and let me briefly draft what we plan to have to accommodate the best Metaverse experience from a chain perspective.
- We plan to have a **Collab feature**, which means, that if you are founding a DAO-like structure with your friends and would like to have a group of 3-5 friends to craft a special featuring, now it's your time to create your boy band. Imagine minting under collective or collecting as collective. This would definitely fuel new social NFTs in a scale, **which could result in DAO2DAO interactions in future.** Example London crew will craft ultimate art with exceptional curation in XR worlds and folks from ItalyDiscoDAO could collect it by infusing their capital deployment.
- Auctions - yes, KodaDot will have live Auctions, especially trustless auctions which could take as much of your audience and not rely on any servers like other promote "their Auctions" which are just hacky way to "proof". We are **working closely with the Basilisk team to leverage their work approved by Web3Foundation right in KodaDot and open flood gates to audience.**
- Royalties - indeed, this is a really anticipated feature we've been missing for a year and was fuel driving revenue for creators' economy. On top of all, we will fuel this with **proxy Royalties**, which **you can stream your revenue** for humanitarian causes from your sales in a trustless way. We even plan to have a bar-fight w @XyloDrone to bit examine why Royalties are super important for Creators Economy and deep-dive on ways you can use it, check our upcoming Twitter Spaces oon
- Make Offers - this is one that slightly opens doors to various Defi applications. In short case, when you list your work, you can tell users to "bid" on your work and you decide which bid you will take in. If you put your art for example on a list and you will get offers for 9 $BSX, 8 $BSX and 7 $BSX, you can freely choose one with 8 BSX if collectors have definitely the greatest reputation for you as an artist, so not need inherently need to take highest one to leave room for price development of your art.
- Revenue Share - This is compounding NFT primitive we plan to have. I.e. You make as Collab sale for 10 $KSM, you can decide what amount of splitting revenue share you would like to see split among the creators group, i.e. what ratio you would like to see, for example, you can split it like **4:3:2:1, i.e idea execut0r, ART crafter, designer, shiller.** You can set it how you would like to drive your splits.
- Verification by holding token - yes, a lot of platforms offer ways to mitigate scams, thus, it's mostly centralized around a single token and we would like in the fashion of many Collabs open doors to other creators groups to emit their own token and swapping between them, similar like you can see Treasury Token Swaps in their all respect and admiration each collab group. I.e. imagine there could be $XYLODRONE and $DAMSKY token, now these two celebrities could create their fandom and they can in mutual admiration ape into each of their works similar to getting their "social" token. What would be even greatest, artists who will craft, you could see their holdings on KodaDot so you can fairly verify their tribe/religion/belief in the particular social tribe. 
- Reactive Actions - this is something we are elaborating further w @vikiival and add there way NFTs could React on their Equipment, similar like NFTs could have their conditions depending on **yield or utility** NFTs are getting in some game &c, it's still open-ended how we will design this, most importantly we are looking ways to game builder open new avenues to craft their games. Are you a game builder? Get in touch with us to influence our development.
- Advanced Traits System aka GameFi-ready - **we learned that lot of NFTs today support only a level 1 traits system**, which isn't bad, but just limits what one can do and of course lot of traits snippers would like to see more entropy. Here we come, we would be allowing NFTs to have nested properties, i.e. if your VRM avatar has tons of objects, sure, our rarity tools can show you how "deep rare" is this avatar and bring deep rarity to the table, just we need to see how best to put it into works.
- Fractions - this is our speciality and their lot of fractionalized or fragmented NFTs in space. We really get the concept yet a lot of these fractionalized NFTs pools are missing liquidity so we are exploring ways how one can farm yield from their NFTs by getting into fractions plus receive extra incentive. This would take action once we will see more expensive NFTs on MetaPrime so take a breathe till then.
- Equip - Really anticipated feature is linking NFTs. In a technical way, we would open people to do one-way or bi-linking. That means you can create a chief-worker scenario, where worker NFT which would be streaming royalties to chief NFT &c and you can think of many fairy tale scenarios, we just have it there.
- Scheduler - You've guessed right that a lot of games need clocks. We are bringing this to timing, i.e. set whenever in your game rewards are distributed based on either block time or external "reality" time. Simple as it sounds, we would enable through ReactiveNFTs enable this for game builders to design and "program" their scenarios
- Protocol Native Licensing - We've met with experts on CreativeCommons who would be advising us on how to best embrace the formability of particular crafts and stream revenues in alignment with licensing
- Forking/Remixing - Okay would that work the same as Github? Yes and no. We noticed that a lot of artists are taking sometimes over 70% of inspiration from the particular artist by losing original references. But what if we can help in a similar manner as in developer circles you can see who is using the origin of your art and how? To avoid "steal" culture, we would like to embrace fork culture to drive respect to origin artworks/game items. Wherewith optionality, imagine setting royalties being streamed. I.e. you can accrue 0.5% for each fork will produce extra money, would be amazing to create the right foundational art, right?

Probably this is just a brief overview of what we are up to, we just need to decide which primitives will take it to MetaPrime v0 test-net (codename SaltMine(?), lot of bugs will be there and we love salt for it's roasty minerals).

The best is to reach out us now to accommodate game designers' and artists' needs so we can reflect market needs best and align our product-market fit and stand on the leading lines of the best NFT infrastructure there. 

**Remember, we are already having the most forked NFT marketplace frontend, crossed 170 forks with 70+ contributors**, we bet we will attract tons of developers for our unique approach and hence we have cemented ourselves as thought leaders on what's needs to be done.

Same as we found other Kusama marketplace effectively been copying our issues and introducing them a few months later as "innovation" we are confirmed we are standing on the edge of innovation in the NFT space 

### MetaPrime Roadmap
Okay, everything sounds cool and superior, what's Roadmap? Yes, this is definitely though one as a lot of work is being delayed on our partner's side and we respect that for various reasons they don't want to rush it. Same as we do, we are mitigating our technical debt on the frontend side in KodaDot. 

You can help us make this fast-forward by joining our culture on Discord. 
We are constantly looking for Vuejs/Typescript, Rust developers and non-technical people help us on Ops and Growth 

### Your turn

We are looking for ways how to open flood gates for any contributors to welcome their input and make Kodadot as a whole a much better place to be and introduce future features. 
I.e. would be a superior nice way to vote on features you would like to see and from there we can see what will be made. This would help us have great user feedback and as we are having incentivized surveys rolling bi-weekly/monthly, this would help us deliver exact that one features requested by audience on granular level

### We are hiring

- Hence my energy levels are resuming I'd be happy to welcome and onboard new members. We are usually looking for smart shadowy super coders to help us solve a few issues faster than ever, thus any related startup position for us as the early stage is super relevant plus for inspiration if your friends are looking here is a list https://kodadot.xyz/jobs/
- Apart from the list we are still looking for folks who would help us on Ops and Growth, writing playbooks for artists, designing ambassadors program etc

**Comment by vikiival** [comment]
**Viki's Meta Minutes**
-

### DEV in numbers:
`is:pr is:merged created:>=2022-03-23`

Since the last MetaHours, so far mode than 80 PRs were merged.
Along with that we introduced a code review system where we assign a person for a code review. We believe that we should scale the operations and therefore people just make a code review and once they approve the PR the they pay the person.

### KodaProtocol:

As @yangwao summarized it really nicely there are couple of other upcoming features:

Bundles - Imagine that @jkdizajn would love to sell 3 KusamaBots (each for 0.3 KSM) but as a bundle he can sell it for 0.6 KSM (so one free :) This opens new possiblities for the gamefi euqipables (so sell the whole armor set) in once.

To the reactivity: 
In simple terms we have a set of rules what should happen with the NFT on the various events. 
So when the NFT is `created` we could instantly list it.
Before the token is `burned` we could fork it into some other generative art based on the 

### Basilisk:

- The indexer is tested and ready for the production
- These days working on the frontend (aka. UI)
- Hope to release a testing version next week (incentivized testing)

**Comment by damskyftw** [comment]
damsky's meta-hours entries

- Announcing the frequent reader program
![Koda_medium__animate_1(1)](https://user-images.githubusercontent.com/92786519/163130680-6185db8d-5aff-4def-a1a6-8727ca6829d5.svg)


- KodaDot's presence in the metaverse
![yyy](https://user-images.githubusercontent.com/92786519/163130558-096dd7cb-1c34-4fab-abd8-957853bc1b82.png)

- Developing ambassador program - let's talk ( will lay out couple ideas during the call ) would be really nice to hear other's feedback.
- Last chance to enter KodaDot survey nr.2 https://form.kodadot.xyz/survey ( Currently 2 spots left )
- Shall we start another contest? meme time :)

Community meetup: irl or offline event

300 $

Article or video in english

150-200$

Article in local language(translation) and activity around that 

100$

Weekly twitter activity -basicly tagging KodaDot and spreading good word 

100-150$
